docs/topics/testing/overview.txt

```diff
@@ -209,10 +209,17 @@ with ability to share the database between threads.
 Order in which tests are executed
 ---------------------------------
 
-In order to guarantee that all ``TestCase`` code starts with a clean database,
-the Django test runner reorders tests in the following way:
+The Django test runner reorders tests in the following way:
 
-* All :class:`~django.test.TestCase` subclasses are run first.
+* To provide the quickest feedback on failures, any failures detected when
+  first loading tests are "run" first. This includes things like test modules
+  that couldn't be found and test modules that couldn't be loaded due to
+  syntax errors.
```

Comment:
I would move this to a separate note, it's an optimization detail that can overwhelm the main section of this paragraph.

Reply:
Also, we could add a small `versionchanged` annotation, e.g.:
```
.. versionchanged:: 4.0

    In older versions, failures detected when loading tests were not ordered
    first.
```